engine/src/main/java/org/hibernate/search/backend/impl/DeletionQueryMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * @author Martin Braun
+ */
+public interface DeletionQueryMapper {
+
+	Query toLuceneQuery(DeletionQuery deletionQuery, ScopedAnalyzer analyzerForEntity);
```

Comment:
> I don't want this stuff in separate places to be honest. fromString(...)
and toString(...) are complete opposites and "undo" each other, so it's a
good idea to keep them in the same place imo.

Hum, ok. It's not uncommon though to have serialization code on the
type-to-be-serialized itself and de-serialization code on some accompanying
factory class (or a static method in said class). I don't really mind
either way, but keep in mind that this approach requires the look-up of the
mapper type which would not be needed otherwise. Would be interesting to
learn what the others think?